public/app/features/datasources/state/actions.ts

```diff
@@ -151,10 +151,15 @@ export function loadDataSourceMeta(dataSource: DataSourceSettings): ThunkResult<
 export async function getDataSourceUsingUidOrId(uid: string | number): Promise<DataSourceSettings> {
   // Try first with uid api
   try {
+    let byUidUri = `/api/datasources/uid/${uid}`;
```

Comment:
nit: maybe create a small util function to do this to prevent the duplication of this code.

Reply:
agreed :+1: